extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -544,23 +544,26 @@ export class AmpA4A extends AMP.BaseElement {
           //    all of the enclosed mutations are fairly simple and unlikely
           //    to fail.
           this.vsync_.mutate(() => {
+            const doc = this.element.ownerDocument;
             // Copy fonts to host document head.
             this.relocateFonts_(creativeMetaData);
-            // Add extensions to head.
-            this.addCreativeExtensions_(creativeMetaData);
-            // Finally, add body and re-formatted CSS styling to the shadow root.
-            const shadowRoot =
-                this.element.shadowRoot || this.element.createShadowRoot();
-            // TODO(dvoytenko, tdrl): Cloning the amp-runtime style from the
-            // host document is a short-term fix.  Ultimately, AMP will provide
-            // a better mechanism for this, and this code will have to be
-            // updated to coordinate with their approach.
-            const style = this.win.document.querySelector(
-                'style[amp-runtime]') ||
-                this.win.document.createElement('style');
-            shadowRoot.appendChild(style.cloneNode(true));
-            // End TODO.
-            shadowRoot./*OK*/innerHTML += (cssBlock + bodyBlock);
+            // Create and setup shadow root.
+            const shadowRoot = createShadowEmbedRoot(this.element,
+                creativeMetaData.customElementExtensions || []);
+            // Add custom CSS.
+            const customStyle = doc.createElement('style');
+            customStyle.setAttribute('amp-custom', '');
+            customStyle.textContent = cssBlock;
+            shadowRoot.appendChild(customStyle);
+            // Add body.
```

Comment:
One question that came up about this while discussing with @keithwrightbos : If there happen to be name collisions between creative-provided CSS and extension-provided CSS, we want to ensure that the behavior is consistent between the 3p iframe rendering pathway and the shadow DOM rendering pathway.

I can't be totally sure from reading this, but it seems like it might be the case that this yields:

```
#shadowRoot
  extensionCSS
  creativeCSS
  creativeBody
```

while the 3p path will have:

```
#shadowRoot
  creativeCSS
  extensionCSS
  creativeBody
```

If that's right, how will the browser interpret  name collisions?  First wins?  Last wins?  Browser-specific?  Something else?  Regardless, would it make sense to arrange it to be the same for both shadow DOM and 3p cases?